conseil-api/src/main/scala/tech/cryptonomic/conseil/api/config/ConseilAppConfig.scala

```diff
@@ -0,0 +1,181 @@
+package tech.cryptonomic.conseil.api.config
+
+import com.github.ghik.silencer.silent
+import com.typesafe.config.{ConfigObject, ConfigValue}
+import com.typesafe.scalalogging.LazyLogging
+import pureconfig.error.{ConfigReaderFailures, ExceptionThrown, FailureReason}
+import pureconfig.generic.ProductHint
+import pureconfig.generic.auto._
+import pureconfig.{ConfigReader, _}
+import scopt.OptionParser
+import tech.cryptonomic.conseil.api.security.Security
+import tech.cryptonomic.conseil.common.config.Platforms._
+import tech.cryptonomic.conseil.api.security.Security._
+import tech.cryptonomic.conseil.common.config._
+
+import scala.util.Try
+
+/** wraps all configuration needed to run Conseil */
+trait ConseilAppConfig {
+
+  type Configurations = (
+      ConseilConfiguration,
+      PlatformsConfiguration,
+      SecurityApi,
+      FailFast,
+      VerboseOutput,
+      MetadataConfiguration,
+      Option[NautilusCloudConfiguration]
+  )
+
+  private case class ArgumentsConfig(
+      failFast: Boolean = false,
+      verbose: Boolean = false
+  )
+
+  /** Lazily reads all configuration upstart, will print all errors encountered during loading */
+  private val argsParser = new OptionParser[ArgumentsConfig]("conseil-api") {
+    opt[Unit]('v', "verbose")
+      .action((_, c) => c.copy(verbose = true))
+      .text("print additional configuration info when the application is launched")
+
+    opt[Unit]('f', "fail-fast")
+      .action((_, c) => c.copy(failFast = true))
+      .text("stops without retrying if the initialization process fails for any reason")
+
+    help('h', "help").text("prints this usage text")
+  }
+
+  protected def loadApplicationConfiguration(commandLineArgs: Array[String]): ConfigReader.Result[Configurations] = {
+    import ConseilAppConfig.Implicits._
+
+    implicit def hint[T]: ProductHint[T] = ProductHint[T](ConfigFieldMapping(CamelCase, CamelCase))
+
+    /** Use the pureconfig convention to handle configuration from the command line */
+    def readArgs(args: Array[String]): ConfigReader.Result[ArgumentsConfig] =
+      argsParser.parse(args, ArgumentsConfig()).toRight[ConfigReaderFailures](sys.exit(1))
+
+    //this extra configuration might be needed as we add send operations to the conseil API
```

Comment:
I would get rid of this, not sure if
a. the write api would pass through conseil-api
b. we'll need to rely on lib-sodium

@vishakh ?